tests/unit/regression/LibSVR_unittest.cc

```diff
@@ -119,12 +118,11 @@ TEST(LibSVR,nu_svr_apply)
 	kernel->init(features_train, features_train);
 
 	LIBSVR_SOLVER_TYPE st=LIBSVR_NU_SVR;
-	LibSVR* svm=new LibSVR(svm_C, svm_nu, kernel, labels_train, st);
-	svm->train();
+	auto svm = std::make_shared<LibSVR>(svm_C, svm_nu, kernel, labels_train, st);
 
 	/* predict */
 	auto predicted_labels =
-	    svm->apply(features_test)->as<RegressionLabels>();
+	    svm->train()->as<LibSVR>()->apply(features_test)->as<RegressionLabels>();
```

Comment:
Could this just be `svm->train()->apply(features_test)->as<RegressionLabels>();`?